design/rich-authorization-requests.md

```diff
@@ -0,0 +1,173 @@
+# Rich Authorization Requests (RAR)
+
+* **Status**: Notes
+* **JIRA**: TBD
+* **Implementation**: https://github.com/tnorimat/keycloak/pull/24 
+
+
+## Motivation
+
+[Rich Authorization Requests][1] is designed to allow OAuth Client to include detailed information to an authorization request. Thus, Client can make a specific authorization requirement.
+
+## Specifications
+
+The spec introduces a new parameter "authorization_details" that allows clients to specify their fine-grained authorization, such as `please let me make a payment with the amount of 45 Euros` or `please give me read access to folder A and write access to file X`.
+
+An example of authorization details using
+the payment authorization object:
+
+````
+[
+      {
+         "type": "payment_initiation",
+         "actions": [
+            "initiate",
+            "status",
+            "cancel"
+         ],
+         "locations": [
+            "https://example.com/payments"
+         ],
+         "instructedAmount": {
+            "currency": "EUR",
+            "amount": "123.50"
+         },
+         "creditorName": "Merchant123",
+         "creditorAccount": {
+            "iban": "DE02100100109307118603"
+         },
+         "remittanceInformationUnstructured": "Ref Number Merchant"
+      }
+   ]
+````
+
+The type of data contained in authorization details is infinite and depends on the field (Payment, Health, Tax Data, etc...)
+
+Therefore, it will be more scalable to add the processing of authorization details as SPI (e.g. `RarProcessor.Class`). Then, an extension will come with the specific implementation.
```

Comment:
The spec should also mention that RAR should work with PAR (Pushed Authorization requests for large authentication_details, as described in the [RAR specification for large requests](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar#section-12.4)

Reply:
Yes, I will add it.